modules/c++/xml.lite/include/xml/lite/Speed.h

```diff
@@ -0,0 +1,341 @@
+     * \brief Adds an XML attribute to an element.
+     *
+     * \param[in] elname   The full name of the element being modified. See
+     *                     setData for syntax.
+     * \param[in] attrname The name of the attribute to add to element elname
+     * \param[in] value    The value of the attribute attrname.If it is not a
+     *                     string, it will be converted internally.
+     */
+    template <typename T>
+      void addAttribute(const std::string &elname,
+                        const std::string &attrname,
+                        T value);
+    /*!
+     * \func setNextData
+     * \brief Creates and sets the next element with the name given at the path
+     *        specified.
+     *
+     *        For example, if the XML document looks like this:
+     *            <root>
+     *                <foo/>
+     *            </root>
+     *        calling  setNextdata("foo","new")  would create another <foo> after
+     *        the currently existing one resulting in:
+     *            <root>
+     *                <foo/>
+     *                <foo>new</foo>
+     *            </root>
+     *
+     * \param[in] elname The full name of the element getting a new sibling.
+     * \param[in] value  The value of the attribute attrname. If value is not
+     *                   a string, it will be converted internally.
+     *
+     * \return A string containing the full name of the new element.
+     */
+    template <typename T>
+      std::string setNextData(const std::string &elname, const T &value);
+
+    /*!
+     * \func declareNamespace
+     * \brief Declares an XML namespace on a given element.
+     *
+     * Note that this method merely declares a namespace. To set the namespace
+     * of elements, see setNamespace()
+     *
+     * \param[in] elname The full name of the element where the namespace will
+     *                    be declared.
+     * \param[in] nsPrefix The namespace prefix being declared.
+     * \param[in] nsURI The URI associated with the namespace.
+     */
+    void declareNamespace(const std::string &elname,
+                          const std::string &nsPrefix,
+                          const std::string &nsURI);
+
+    /*!
+     * \func setNamespace
+     * \brief Sets a namespace prefix on an element and possibly all children.
+     *
+     * The recurse option is true by default.
+     * The element will be created if it does not already exist, but don't use
+     * this function for that. If there are going to be child nodes also in
+     * this namespace, consider creating them all using setData, then using
+     * setNamespace to set them all at once. Calling setNamespace for leaf
+     * nodes (as in the case where the element does not yet exist) with
+     * recursion enabled will perform a needless search of the existing nodes.
+     *
+     * \param[in] elname The full name of the element being modified.
+     * \param[in] nsPrefix 
+     */
```

Comment:
Add documentation for `recurse`.  Should have spaces between equals sign (`bool recurse = true`) and don't need `const` since you're passing it in by value